library/src/blas3/rocblas_trsm.hpp

```diff
@@ -0,0 +1,1737 @@
+                                                                 m,
+                                                                 stride_X,
+                                                                 batch_count);
+                        if(i - BLOCK < 0)
+                            break;
+                        rocblas_gemm_template<BATCHED, !BATCHED>(handle,
+                                                                 transB,
+                                                                 transA,
+                                                                 m,
+                                                                 i,
+                                                                 BLOCK,
+                                                                 &negative_one<T>,
+                                                                 0,
+                                                                 (U)X,
+                                                                 i * m,
+                                                                 m,
+                                                                 stride_X,
+                                                                 A,
+                                                                 i * lda + offset_Ain,
+                                                                 lda,
+                                                                 stride_A,
+                                                                 &one<T>,
+                                                                 0,
+                                                                 B,
+                                                                 offset_Bin,
+                                                                 ldb,
+                                                                 stride_B,
+                                                                 batch_count);
+                    }
+                }
+            }
+        } // tranpsose
+
+        return rocblas_status_success;
+    }
+
+    template <rocblas_int BLOCK, bool BATCHED, typename T, typename U, typename V>
+    rocblas_status special_trsm_template(rocblas_handle    handle,
+                                         rocblas_side      side,
+                                         rocblas_fill      uplo,
+                                         rocblas_operation transA,
+                                         rocblas_diagonal  diag,
+                                         rocblas_int       m,
+                                         rocblas_int       n,
+                                         const T*          alpha,
+                                         rocblas_stride    stride_alpha,
+                                         U                 A,
+                                         rocblas_int       offset_Ain,
+       
```

Comment:
I expect the cast type for x_temparr/x_temp in these 2 spots to be V since they both hold the output values.

Reply:
On line 1719 I'm essentially casting `const` to a non-`const` value since the copy function is expected a const value, so I think it's fine there. I'm not sure which other spot you're referring to.

Reply:
maybe a inline comment to improve readability,